pkg/capabilities/capabilities.go

```diff
@@ -245,15 +244,10 @@ outerLoop:
 			}
 
 			vs = append(vs, response.Value)
-
 		// Timeout when a capability panics, crashes, and does not close the channel.
 		case <-ctxWithT.Done():
+			// Not necessarily, the context could have been cancelled by the caller
```

Comment:
Correct -- could you modify the comment to clarify that this also applies if the context is canceled?